c2rust-analyze/tests/filecheck/catch_panic.rs

```diff
@@ -0,0 +1,29 @@
+// This test deliberately exercises unsupported code to check that c2rust-analyze can recover from
+// panics during the analysis.
+
+use std::ptr::NonNull;
+
+// CHECK: final labeling for "good"
+unsafe fn good(p: *mut u8) {
+    *p = 1;
+}
+
+// Analysis of `bad` should fail because it calls an unsupported library function involving raw
+// pointers, namely `NonNull::as_ptr`.
+// CHECK-NOT: final labeling for "bad"
+unsafe fn bad(p: NonNull<u8>) {
+    *p.as_ptr() = 1;
+}
+
+// Analysis of `bad2` should also fail because it has a callee on which analysis failed.
+// CHECK-NOT: final labeling for "bad2"
+unsafe fn bad2(p: NonNull<u8>) {
```

Comment:
Thoughts on adding a test asserting the functionality we know to be broken, e.g. that `good2` defined before `bad3` where `bad3` calls `good2`, will not produce any of the error messages below? That way we can develop against this test case later on

Reply:
I added a new `bad_call_good` function, which fails analysis and also calls `good`, and a check that this does not result in the failure propagating from `bad_call_good` to its callee `good`.  But I'm not sure if that's what you were suggesting.